helix-term/src/ui/prompt.rs

```diff
@@ -683,18 +695,26 @@ impl Component for Prompt {
     }
 
     fn render(&mut self, area: Rect, surface: &mut Surface, cx: &mut Context) {
-        self.render_prompt(area, surface, cx)
+        self.render_prompt(area, surface, cx);
     }
 
     fn cursor(&self, area: Rect, editor: &Editor) -> (Option<Position>, CursorKind) {
+        let area = area
+            .clip_left(self.prompt.len() as u16)
+            .clip_right(if self.prompt.len() > 0 { 0 } else { 2 });
```

Comment:
this `if self.prompt.len()` is mainly here to distinguish between the command prompt (opened with `:`) and other pickers. 

Pickers have a count information at the end of the prompt that I need to take into account if I want the cursor to be correctly displayed. Didn't find any ways to do the distinction except using this prompt. I might leave a comment in the code to explain that if I don't find any better way to handle this